velox/functions/prestosql/tests/ArrayAnyMatchTest.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+class ArrayAnyMatchTest : public functions::test::FunctionBaseTest {
+ protected:
+  // Evaluate an expression.
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const VectorPtr& input) {
+    auto result = evaluate(expression, makeRowVector({input}));
+    assertEqualVectors(expected, result);
+  }
+
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const RowVectorPtr& input) {
+    auto result = evaluate(expression, (input));
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayAnyMatchTest, basic) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{std::nullopt, 2, 3},
+       {-1, 3},
+       {2, 3},
+       {},
+       {std::nullopt, std::nullopt}});
+  auto expectedResult =
+      makeNullableFlatVector<bool>({true, true, true, false, std::nullopt});
+  testExpr(expectedResult, "any_match(c0, x -> (x > 1))", input);
+  expectedResult = makeFlatVector<bool>({true, false, false, false, true});
+  testExpr(expectedResult, "any_match(c0, x -> (x is null))", input);
+
+  input = makeNullableArrayVector<bool>(
+      {{false, true},
+       {false, false},
+       {std::nullopt, true},
+       {std::nullopt, false}});
+  expectedResult =
+      makeNullableFlatVector<bool>({true, false, true, std::nullopt});
+  testExpr(expectedResult, "any_match(c0, x -> x)", input);
+
+  auto emptyInput = makeArrayVector<int32_t>({{}});
```

Comment:
It looks like this is already tested: L48

Reply:
Removed